### ISSUE TITLE
space-before-function-parentheses false positive on generator methods

### ISSUE THREAD

**Issue by steadicat** [eslint/eslint#2082]
```javascript
// test.js
class Foo {
  *bar() {
  }
}
```
```javascript
// config.json
{
    "ecmaFeatures": {
        "generators": true,
        "classes": true
    },
    "rules": {
        "space-before-function-parentheses": [2, "never"]
    }
}
```
```
$ eslint --reset --no-eslintrc --config confing.json test.js

test.js
  2:10  error  Unexpected space before function parentheses  space-before-function-parentheses
```

This is similar to #1929, but not quite the same.

**Comment by btmills** [comment]
When the fix for #1929 was implemented, Espree didn't handle generator methods yet, so we should be unblocked on this now.

**Comment by btmills** [comment]
Working on this. It'll depend on #2050, so the fix will go into `space-before-function-parens`.

**Comment by nzakas** [comment]
Any chance to get the change in before #2050? I'd like to do a patch release today or tomorrow and would like to get this fix it.

**Comment by btmills** [comment]
Yep, #2083.

**Comment by nzakas** [comment]
You are my hero.
